ui/js/models/resultset.js

```diff
@@ -269,35 +271,66 @@ treeherder.factory('ThResultSetModel', ['$rootScope', '$http', '$location',
                     });
 
                     return $q.all([
-                        $q.resolve().then(function () {
+                        $q.resolve().then(() => {
                             if (bbnames.length === 0) {
                                 return;
                             }
+
                             let bbdata = {
                                 "requested_jobs": bbnames,
                                 "decision_task_id": decisionTaskID
                             };
+
                             return $http.post(
                                 thUrl.getProjectUrl("/resultset/", repoName) + resultset_id + '/trigger_runnable_jobs/',
                                 bbdata
                             );
                         }),
-                        $q.resolve().then(function () {
+                        $q.resolve().then(() => {
                             if (tclabels.length === 0) {
                                 return;
                             }
-                            let url = queue.buildSignedUrl(queue.getLatestArtifact, decisionTaskID, 'public/action.yml');
-                            return $http.get(url).then(function (resp) {
-                                let action = resp.data;
-                                let template = $interpolate(action);
-                                let taskLabels = tclabels.join(',');
-                                action = template({
-                                    action: 'add-tasks',
-                                    action_args: `--decision-id ${decisionTaskID} --task-labels ${taskLabels}`,
+
+                            let actionTaskId = tc.slugid();
+                            let url = queue.buildSignedUrl(queue.getLatestArtifact, decisionTaskID, 'public/actions.json');
+
+                            return $http.get(url).then((resp) => {
+                                // Use action.yml if it's the wrong version of actions.json
+                                if (resp.data.version !== 1) {
```

Comment:
Do you have an example of this particular case I could use to test? Would `resp.data` be set to `{}` or `undefined` in that case? If it's the latter, then I could do:

```
if (!resp.data || resp.data.version !== 1) {
  ...
}
```